web/src/beta/features/ProjectSettings/innerPages/PublicSettings/AliasSettings/EditPanel.tsx

```diff
@@ -7,39 +7,83 @@ import {
   Typography
 } from "@reearth/beta/lib/reearth-ui";
 import { CommonField } from "@reearth/beta/ui/fields";
+import {
+  useProjectFetcher,
+  useStorytellingFetcher
+} from "@reearth/services/api";
 import { useT } from "@reearth/services/i18n";
-import { useNotification } from "@reearth/services/state";
 import { styled, useTheme } from "@reearth/services/theme";
-import { FC, useCallback, useState } from "react";
+import { FC, useCallback, useMemo, useState } from "react";
 
-import { AliasSettingProps } from ".";
+type Prop = {
+  publicUrl: string;
+  isStory?: boolean;
+  alias?: string;
+  itemId?: string;
+  onClose?: () => void;
+  onSubmit?: (alias?: string) => void;
+};
 
-const EditPanel: FC<AliasSettingProps> = ({
+const EditPanel: FC<Prop> = ({
   alias,
   isStory,
+  publicUrl,
+  itemId,
   onClose,
   onSubmit
 }) => {
   const t = useT();
   const theme = useTheme();
-  const [localAlias, setLocalAlias] = useState("");
-  const [warning, setWaring] = useState(false);
-  const [, setNotification] = useNotification();
-
-  const handleChange = useCallback(
-    (value: string) => {
-      if (value === alias) {
-        setWaring(true);
-        setNotification({
-          type: "error",
-          text: isStory
-            ? t("Unable to update published story alias.")
-            : t("Unable to update published project alias.")
-        });
-      } else setWaring(false);
-      setLocalAlias(value);
-    },
-    [alias, isStory, setNotification, t]
+  const { checkProjectAlias } = useProjectFetcher();
+  const { checkStoryAlias } = useStorytellingFetcher();
+
+  const [prefix, suffix] = useMemo(() => {
+    if (!publicUrl) return ["", ""];
+    const aliasMatch = publicUrl.match(/^(https?:\/\/[^?]+)\?alias=/);
+    if (aliasMatch) {
+      return [aliasMatch[1] + "?alias=", ""];
+    }
+
+    const subdomainMatch = publicUrl.match(/^https?:\/\/([^\\.]+)\.(.+)$/);
+    if (subdomainMatch) {
+      return ["https://", "." + subdomainMatch[2]];
+    }
+
+    return ["", ""];
+  }, [publicUrl]);
```

Comment:
Could we re-use this https://github.com/reearth/reearth-visualizer/blob/6cfc87e436a37983270cf4647274fdc16fa80f87/web/src/beta/features/ProjectSettings/innerPages/PublicSettings/AliasSettings/index.tsx#L57?
I think it's better to keep using one single util function to get the prefix and suffix.